lib/dry/auto_inject/builder.rb

```diff
@@ -13,13 +13,16 @@ class Builder
       def initialize(container, options = {})
         @container = container
         @strategies = options.fetch(:strategies) { Strategies }
+      end
+
+      def method_missing(name, *args, &block)
+        return super unless strategies.key?(name)
+
+        Injector.new(container, strategies[name])
+      end
```

Comment:
Yeah, you're right @solnic, that if/else is much clearer for a method with only 2 single-line pathways :)